PyInstaller/hooks/hook-enchant.py

```diff
@@ -18,3 +20,24 @@
     for d in files:
         for f in d[1]:
             datas.append((f, d[0]))
+elif is_darwin:
+    # Collecting dictionaries only works when pyenchant is installed via pip since the dictionaries installed via pip
+    # are stored next to the enchant library
+    binaries = []
+    files = exec_statement("""
+from enchant._enchant import e
+print(e._name)""").strip().split()
+    for f in files:  # Put the enchant library in lib/ so the enchant plugins can find it
+        binaries.append((os.path.join('lib', os.path.basename(f)), f))
```

Comment:
PyEnchant can find the enchant.dylib anywhere as long as we add the path it's in to the runtime hook. The problem comes with the enchant library and its plugins. The enchant library can only find its plugins if they are in the proper directory relative to it, e.g., `lib/enchant/libenchant_myspell.so` and the plugins can only find the enchant library if they are in the proper directory relative to it, e.g., `lib/libenchant.1.dylib`.

This should work with macports and homebrew but I haven't tested due to the dirtionary issue.